### PR TITLE
Fix CA1806

### PR DESCRIPTION
**Pull request by gintsk**
# Pull Request

## The issue or feature being addressed
Contributes to #1290

## Details on the issue fix or feature implementation
Formal fix for CA1806: Do not ignore method results

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: test/Polly.Specs/Caching/AbsoluteTtlSpecs.cs [hunk 1]
```diff
@@ -6,23 +6,32 @@ public class AbsoluteTtlSpecs : IDisposable
     [Fact]
     public void Should_be_able_to_configure_for_near_future_time()
     {
-        Action configure = () => new AbsoluteTtl(DateTimeOffset.UtcNow.Date.AddDays(1));
```
Comment (martincostello) [inline test/Polly.Specs/Caching/AbsoluteTtlSpecs.cs:9]: An easier way to fix all of these issues without also changing all the formatting is to assign them to the discard variable, i.e. ` _ = new Foo()` etc.

Comment (martincostello) [reply test/Polly.Specs/Caching/AbsoluteTtlSpecs.cs:9]: I meant more like this, assuming it compiles:

```csharp
Action configure = () => _ = new AbsoluteTtl(DateTimeOffset.UtcNow.Date.AddDays(1));
```

Then the diff is mostly just adding a bunch of `_ = `, than also changing the lines, adding braces etc.

Comment (gintsk) [reply test/Polly.Specs/Caching/AbsoluteTtlSpecs.cs:9]: Agree. This will be cleaner.

**Review by martincostello — APPROVED** [review]

**Comment by martincostello** [comment]
Thanks again!

File: test/Polly.Specs/Caching/AbsoluteTtlSpecs.cs (hunk 1 above)
Comment (jnyrup) [reply test/Polly.Specs/Caching/AbsoluteTtlSpecs.cs:9]: Or if you permit using `var` and C#10, we can let the compiler infer the return type to be `Func<AbsoluteTtl>` which also silences CA1806 as I recall.

```cs
var configure = () => new AbsoluteTtl(DateTimeOffset.UtcNow.Date.AddDays(1));
```

**Pull request merged**